[Bug][Connector-V2][Email] Fix NPE on null values, add configurable attachment name and delimiter

zhan7236 opened the pull request:
### Purpose of this pull request

Fixes three issues in Email Sink connector (apache/seatunnel#9381):
1. NPE when upstream field values are null
2. Emails sent even with no data written
3. Hardcoded attachment filename and field delimiter

### Does this PR introduce _any_ user-facing change?

Yes. Adds two new optional configuration parameters with backward-compatible defaults:

```yaml
sink:
  Email:
    email_attachment_name: "custom-report.csv"  # default: "emailsink.csv"
    email_field_delimiter: "|"                  # default: ","
```

**Previous behavior:**
- Null field values caused `NullPointerException`
- Empty emails sent when no data written
- Attachment filename fixed as "emailsink.csv"
- Field delimiter fixed as comma

**New behavior:**
- Null fields render as empty strings
- Email skipped if no data written (logged as "No data to send, skipping email")
- Attachment filename and delimiter configurable via options above

### How was this patch tested?

Added unit tests in `EmailSinkWriterTest`:
- `testWriteWithNullValues()` - verifies null handling without NPE
- `testCustomDelimiter()` - verifies custom configuration options

All existing tests pass.

### Check list

* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If necessary, please update `incompatible-changes.md` to describe the incompatibility caused by this PR.
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)

zhan7236 committed:
[Bug][Connector-V2][Email] Fix NPE on null values, add configurable attachment name and delimiter

zhan7236 committed:
trigger ci

zhangshenghang commented:
Please supplement the document: https://seatunnel.apache.org/zh-CN/docs/2.3.12/connector-v2/sink/Email

zhan7236 committed:
docs: supplement Email sink connector documentation for new config options

Add documentation for email_attachment_name and email_field_delimiter configuration options in both Chinese and English versions.

zhan7236 commented:
Hi @zhangshenghang,

I've supplemented the documentation for both Chinese and English versions:
- docs/zh/connector-v2/sink/Email.md
- docs/en/connector-v2/sink/Email.md

Added documentation for the two new configuration options:
- email_attachment_name: The name of the email attachment file (default: emailsink.csv)
- email_field_delimiter: The delimiter used to separate fields (default: ,)

Please review. Thanks!

davidzollo reviewed (approved):
+1
Good job
